### PR TITLE
Add pvr.sledovanitv.cz addon

### PR DESCRIPTION
**Pull request by palinek**
Hi,
 this is client for the Czech IP TV provider https://sledovanitv.cz (works also for Slovakian subscribers - https://sledovanietv.sk).

### CODE REVIEW

File: pvr.sledovanitv.cz/pvr.sledovanitv.cz.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+pvr.sledovanitv.cz https://github.com/palinek/pvr.sledovanitv.cz.git master
```
Comment (Rechi) [inline pvr.sledovanitv.cz/pvr.sledovanitv.cz.txt]: `pvr.sledovanitv.cz https://github.com/palinek/pvr.sledovanitv.cz master`

**Comment by Rechi** [comment]
@ksooo please have a look at this

File: pvr.sledovanitv.cz/pvr.sledovanitv.cz.txt (hunk 1 above)
Comment (palinek) [reply pvr.sledovanitv.cz/pvr.sledovanitv.cz.txt]: Updated..

**Comment by palinek** [comment]
Is there something to do on my side here?

**Comment by ksooo** [comment]
Yes, please stay patient. We're doing all Kodi work as volunteers in our free time, real life has usually priority.

**Comment by palinek** [comment]
> Yes, please stay patient.

I'm OK :)... I just wanted to be sure, that you're not waiting on some action I should make.

**Comment by PepaRokos** [comment]
Hi, have you any support from sledovanitv company or backend developers? They stoped supporting me some time ago and this is the reason why my repository (https://github.com/PepaRokos/pvr.sledovanitv) is not here in offical addons. Unofficial and unsupported addons shuld stay outside of this repo in my opinion.

**Comment by ksooo** [comment]
>  They stoped supporting me some time ago

 What does this mean? What did they do for you before they stopped?

 For me, "supported addon" means that the addon itself has an active maintainer who cares to keep the addon running.

**Comment by ksooo** [comment]
> Unofficial and unsupported addons shuld stay outside of this repo in my opinion.

 I agree to that unsupported addons must not be part of this repo.

 Addon's state changes from "unofficial" to "official" exactly be including it in this repo. So, only solution to get the "official" state is to include it here. ;-)

**Comment by PepaRokos** [comment]
They does not have public API so gave me acces to theyr documentation. Now emails about new version of API left unanswered. I mean this addon is little bit unwanted from theyr site.

**Comment by ksooo** [comment]
> They does not have public API so gave me acces to theyr documentation. Now emails about new version of API left unanswered. I mean this addon is little bit unwanted from theyr site.

 Thanks for the headsup. In this case, sorry, no inclusion of this addon in this repo.

**Comment by ksooo** [comment]
>  I mean this addon is little bit unwanted from theyr site.

Sorry, but we do not want to run into trouble with the content provider.

**Comment by palinek** [comment]
> They does not have public API

The API is public... but there is no public documentation for it. But there is no problem to trace the needed calls and reimplement it.

>> I mean this addon is little bit unwanted from theyr site.

> Sorry, but we do not want to run into trouble with the content provider.

I see nothing wrong to use the very same API as the "native" application as long the end user pays the service. And that's the case of this addon...we don't steal anything, we're just using the content/service, that we've paid for.

**Comment by ksooo** [comment]
> I mean this addon is little bit unwanted from theyr site.

 Clarify with them that your addon is okay for them, prove that to us and we are in business.

**Comment by palinek** [comment]
... and to quote the general commercial conditions https://sledovanitv.cz/info/s/vop:

> 4.1.Uživatel je povinen si zabezpečit hardwarové i softwarové prostředky včetně jejich instalace, vstupní a výstupní zařízení, včetně televizního přijímače (dále jen „Zařízení Uživatele“) a jejich funkčnost a kompatibilitu se Službou IPTV tak, aby bylo možné připojení ke koncovému bodu sítě Provozovatele. Provozovatel neodpovídá za funkčnost či provoz těchto Zařízení Uživatele. Připojení jiných zařízení než zařízení otestovaných Provozovatelem je možné jen na výlučnou odpovědnost Uživatele a pouze za předpokladu, že takové zařízení je určeno a řádně certifikováno pro trh Evropské unie. Seznam Provozovatelem otestovaných zařízení (včetně softwarových řešení) je uveden na internetových stránkách. V případě připojení jiných zařízení než schválených Provozovatelem nenese Provozovatel odpovědnost za funkčnost a kvalitu Služby IPTV a v případě nefunkčnosti, nedostupnosti či snížení dostupnosti Služby IPTV negarantuje Provozovatel v těchto případech úroveň poskytovaných služeb.

English (google translate):

> 4.1.The user is obliged to supply the hardware and software resources including their installation, input and output devices, including the television set (hereinafter referred to as the "User's Device") and their functionality and compatibility with the IPTV Service in order to connect to the endpoint of the Operator's network . The operator is not responsible for the functionality or operation of these User's Devices. The connection of devices other than equipment tested by the Operator can only be the sole responsibility of the User and only if such equipment is designated and properly certified for the European Union market. The list of devices tested by Operator (including software solutions) is listed on the website. In the event of connections other than those approved by the Operator, the Operator shall not be liable for the functionality and quality of the IPTV Service, and in the event of the IPTV Service being rendered inoperable, unavailable or diminished, the Operator shall not guarantee the level of service provided.

So this is crystal clear to me: "I can use whatever device I want and it is my responsibility. But in case using a 3rd party solution the provider doesn't take any responsibility for bad service."

By stating this I don't see any blocker to provide "official kodi" addon for this content provider as long it is working and people want to use it.

**Comment by palinek** [comment]
> Clarify with them that your addon is okay for them, prove that to us and we are in business.

Is quoting of official conditions enough?

**Comment by palinek** [comment]
> > Clarify with them that your addon is okay for them, prove that to us and we are in business.

> Is quoting of official conditions enough?

..and this is a screenshot of the website with the "xbmc" device included:
screenshot

**Comment by ksooo** [comment]
Please understand that we as a small non-profit hobby project cannot afford to get in trouble with any commercial content provider. Not blocking your personal "xbmc" device does not mean that they are fine with what you are doing. Go, write them an email, asking for their permission to use your client for their services. Once we've seen their okay, we can include your addon in our offial repo.

**Comment by palinek** [comment]
> Not blocking your personal "xbmc" device does not mean that they are fine with what you are doing.

No, but the commercial conditions state it without any doubt. Every user is responsible for setting up his/her device. So as long, as it works and end user conditions aren't changed ... why couldn't we/you provide a way for people to use their paid service?

**Comment by ksooo** [comment]
If you are so sure about being legit, why don't you just get in touch with the content provider and ask for permission? Then, we are all on the safe side.

**Comment by ksooo** [comment]
> why couldn't we/you provide a way for people to use their paid service?

 We can, but only if we are 100% (!) sure not to do illegal things. As long as there is any doubt, no inclusion in this repo. Remember, we are not making a single dime out of this, but risk to loose reputation/money!

**Comment by PepaRokos** [comment]
@ksooo You should know I am confused with this rework itself. It produces many useless requests to provider's servers (approx 30x more per minute then my version) - recordings are loaded every second probably for reflect changes made outside of Kodi (every 5 minutes could by acceptable). EPG loading does not reflect Kodi PVR settings (time frame and sync interval), instead of it chunks of EPG are loaded every minute. Ones a day "full sync" is performed which is not realy full to reflect Kodi settings. EPG loading is already solved in my repo in simple way. Channel list is reloaded ones a day before EPG "full sync", this is nonsense probably caused by misunderstood provider's API  (I have old documentation but have any, palinek has only assumption from request tracing). This are reasons why I reject pull request to my repo. Please consider this before you make final decision.

**Comment by palinek** [comment]
@PepaRokos 

> It produces many useless requests to provider's servers (approx 30x more per minute then my version)

```
$ egrep 'CurlFile::Open.*https://sledovanitv.cz' .kodi/temp/kodi.log 
21:06:09.897 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70002980) https://sledovanitv.cz/api/device-login?deviceId=1560857838&password=sjgq14mws7rfz646lkm3
21:06:11.124 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f700f9450) https://sledovanitv.cz/api/playlist?format=androidtv&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:11.593 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f700be700) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:11.748 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f700e0db0) https://sledovanitv.cz/api/record-timeshift?format=m3u8&recordId=32922591&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:11.898 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f7007b950) https://sledovanitv.cz/api/record-timeshift?format=m3u8&recordId=32228867&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:12.071 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70053c20) https://sledovanitv.cz/api/record-timeshift?format=m3u8&recordId=32228862&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:12.213 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f700cbf80) https://sledovanitv.cz/api/record-timeshift?format=m3u8&recordId=32099827&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:12.385 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f7011c480) https://sledovanitv.cz/api/epg?detail=1&duration=60&time=2018-04-18+21%3a06&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:12.677 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70107dd0) https://sledovanitv.cz/api/epg?detail=1&duration=1439&time=2018-04-18+22%3a06&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:16.805 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70b08180) https://sledovanitv.cz/api/epg?detail=1&duration=1439&time=2018-04-17+21%3a06&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:20.984 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70d3f390) https://sledovanitv.cz/api/epg?detail=1&duration=1439&time=2018-04-19+22%3a06&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:24.927 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f71976320) https://sledovanitv.cz/api/epg?detail=1&duration=1439&time=2018-04-16+21%3a06&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:30.424 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f71883020) https://sledovanitv.cz/api/epg?detail=1&duration=1439&time=2018-04-15+21%3a06&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:50.400 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f72e16d20) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:06:51.509 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70103610) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:07:10.675 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f72edc010) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:07:12.008 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70ba5fe0) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:07:31.269 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f73184880) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:07:32.358 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f731e76a0) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:07:50.501 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70eb7cc0) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:07:51.705 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f73108780) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:08:10.817 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f72f1be70) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:08:12.000 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f72f3fc40) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:08:31.164 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f7316e610) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:08:32.258 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f71462260) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:08:50.407 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f7019ffd0) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:08:51.495 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f72e03b50) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:09:10.666 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70b88230) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:09:11.854 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f702aef60) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:09:30.993 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70b900c0) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:09:32.207 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f7075f310) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:09:50.369 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f730e1fe0) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:09:51.475 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70b851b0) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:10:10.607 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f70103610) https://sledovanitv.cz/api/keepalive?&PHPSESSID=h92qa47327m12sq9c0flshpo80
21:10:11.802 T:140254651537152   DEBUG: CurlFile::Open(0x7f8f72edc010) https://sledovanitv.cz/api/get-pvr?&PHPSESSID=h92qa47327m12sq9c0flshpo80
```
=> initial login, getting channel/group list, record list (and stream url for each recording) & EPG, then each 20 seconds a "keep-alive" call and a record list. Nothing more.

> recordings are loaded every second probably for reflect changes made outside of Kodi (every 5 minutes could by acceptable). 

The above log (the `sledovanitv.cz/api/get-pvr` calls) clearly states, this is not the case. Sure this loading can be done more or less frequently based on each user preferences.

> EPG loading does not reflect Kodi PVR settings (time frame and sync interval), instead of it chunks of EPG are loaded every minute.

The addon is reading EPG in chunks: first (after start) only current hour to provide the information for user as quick as possible, next chunks are read by days. Again the log proves that `chunks of EPG are loaded every minute` is a lie (the `sledovanitv.cz/api/epg` calls). 

> Ones a day "full sync" is performed which is not realy full to reflect Kodi settings.

The full EPG refresh is done once a day just because of "EPG tag availability refresh" and only once a day to save the needed calls to provider. (not doing the full refresh of all already read information on each Kodi's refresh -> in this case only not yet read timestamps are gathered).

> Channel list is reloaded ones a day before EPG "full sync", this is nonsense probably caused by misunderstood provider's API

The channel list is reloaded because:
1. it is a simple call done once a day
2. order of channels can be changed outside of kodi and IMO kodi should reflect that (of course the periodicity of such refresh can be pulled to user settings)

> This are reasons why I reject pull request to my repo. 

This is the funniest lie.... in the mentioned PR you were arguing about synchronization, using new features of C++11 and using a thread for HTTP communication to provider's server.

All your comments here seems to me that you're acting like a little child that misses its toy. The code is open, every one can review it and it seems you're lying here intentionally.

I have no problem with you rejecting to include my changes. I also do not insist in including my recipe in Kodi's official repository. I'll happily use it by my own.

I just wanted to share it to the people as I like FOSS and really don't see a reason why it shouldn't be "official" to get it to the people. I'm also happy if your addon is included here instead.

**Comment by ksooo** [comment]
I will unsubscribe now from this PR.

**Comment by palinek** [comment]
> I will unsubscribe now from this PR.

Good choice. So am doing I :)

**Pull request closed without merging**